### ISSUE TITLE
Improve Host connection handling

### ISSUE THREAD

**Issue by quietlychris** [quietlychris/rhiza#3]
This now, the Host handles incoming connection by spawning a new `tokio` process per incoming TcpStream, which are then handled in an thread-safe `Vec<JoinHandle<()>>` data structure called `connections`. However, this limits the information that the Host has once the connection has been established which may be of interest, including information like the peer address or other information that may be gathered during the initial connection handshake. This may also be useful in the future to adding support for encrypted connections, unique connection names, or supporting incoming streams from multiple interfaces.

One possible way to handle this may be to create a new `Connection` struct that contains the `JoinHandle`, rather than simply operating on the handle itself. One potential option could be something like: 
```rust
pub struct Connection {
    handle: JoinHandle<()>
    peer_addr: SocketAddr,
    connection_name: String
}
```

**Comment by quietlychris** [comment]
Partially addressed in #4

**Comment by quietlychris** [comment]
Partially addressed in #6
